lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2131,6 +2131,13 @@ func (d *zfs) MigrateVolume(vol Volume, conn io.ReadWriteCloser, volSrcArgs *mig
 		}
 	}
 
+	// If we haven't negotiated zvol support, ensure volume is not a zvol.
+	if !shared.StringInSlice(migration.ZFSFeatureZvolFilesystems, volSrcArgs.MigrationType.Features) {
+		if d.isBlockBacking(vol) {
```

Comment:
Any particular reason why you've used a nested if here rather than an `&&` on the if above?